File: app/src/main/java/sgtmelon/scriptum/office/intf/MenuIntf.java
package sgtmelon.scriptum.office.intf;

public interface MenuIntf {

    /**
     * Интерфейс для меню внутри заметки
     */
    interface Note {

        interface DeleteMenuClick {
            void onMenuRestoreClick();

            void onMenuRestoreOpenClick();

            void onMenuClearClick();

            void onMenuDeleteClick();
        }

        interface NoteMenuClick {
            /**
             * @param editModeChange - Надо ли менять режим редактирования
             * @param showToast      - Показывать уведомление о том, что заметка пуста
             * @return - Сохранение успешно при возвращении true
             */
            boolean onMenuSaveClick(boolean editModeChange, boolean showToast);

            void onUndoClick();

            void onRedoClick();

            void onMenuRankClick();

            void onMenuColorClick();

            /**
             * @param editMode - Установка режима редактирования
             */
            void onMenuEditClick(boolean editMode);

            void onMenuCheckClick();

            void onMenuBindClick();

            void onMenuConvertClick();
        }

    }

    /**
     * Интерфейс для меню при долгом нажатии на заметку
     */
    interface Dialog {

        interface DeleteMenuClick {
            void onMenuRestoreClick(int p);

            void onMenuCopyClick(int p);

            void onMenuClearClick(int p);
        }

        interface NoteMenuClick {
            void onMenuCheckClick(int p);

            void onMenuBindClick(int p);

            void onMenuConvertClick(int p);

            void onMenuCopyClick(int p);

            void onMenuDeleteClick(int p);
        }

    }

}

File: app/src/main/java/sgtmelon/scriptum/office/intf/MenuIntf.kt
package sgtmelon.scriptum.office.intf

interface MenuIntf {

    /**
     * Интерфейс для меню внутри заметки
     */
    interface Note {

        interface DeleteMenuClick {
            fun onMenuRestoreClick()

            fun onMenuRestoreOpenClick()

            fun onMenuClearClick()

            fun onMenuDeleteClick()
        }

        interface NoteMenuClick { // TODO: 10.12.2018 добавить длинное нажатие на кнопку undo/redo - для возвращение в один из концов
            /**
             * При успешном сохранении возвращает - true
             * [modeChange] - Надо ли менять режим редактирования,
             * [showToast]  - Показывать уведомление о том, что заметка пуста
             */
            fun onMenuSaveClick(modeChange: Boolean, showToast: Boolean): Boolean

            fun onUndoClick()

            fun onRedoClick()

            fun onMenuRankClick()

            fun onMenuColorClick()

            /**
             * [editMode] - Установка режима редактирования
             */
            fun onMenuEditClick(editMode: Boolean)

            fun onMenuCheckClick()

            fun onMenuBindClick()

            fun onMenuConvertClick()
        }

    }

}